Sources/KSCrashRecording/include/KSCrashError.h

```diff
@@ -0,0 +1,66 @@
+//
+//  KSCrashError.h
+//
+//  Created by Gleb Linnik on 12.07.2024.
+//
+//  Copyright (c) 2012 Karl Stenerud. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall remain in place
+// in this source code.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef KSCrashError_h
+#define KSCrashError_h
+
+#ifdef __OBJC__
+#include <Foundation/Foundation.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef __OBJC__
+static NSErrorDomain const KSCrashErrorDomain = @"KSCrashErrorDomain";
+#endif
+
+typedef
+#ifdef __OBJC__
+    NS_ERROR_ENUM(KSCrashErrorDomain, KSCrashInstallErrorCode)
+#else  /* __OBJC__ */
+    enum
+#endif /* __OBJC__ */
+{ KSCrashInstallErrorAlreadyInstalled = 1,
```

Comment:
You can call it `KSCrashInstallResultCode` and have `KSCrashInstallResultCodeSuccess = 0`.
In this case `NS_ERROR_ENUM` is not fully accurate option but at least C API would not return `0` that is not a part of this enum.

Reply:
About `KSCrashInstallResultCode`. In the Swift interface it's called `KSCrashInstallResult`:

```swift
public struct KSCrashInstallResult : CustomNSError, Hashable, Error {
```

Maybe we should just add `KSCrashInstallErrorNone = 0`?